ui/src/tempVars/components/EditTemplateVariable.tsx

```diff
@@ -0,0 +1,271 @@
+        isNew: true,
+      }
+    }
+  }
+
+  public render() {
+    const {source, onCancel} = this.props
+    const {nextTemplate, isNew} = this.state
+    const TemplateBuilder = this.getTemplateBuilder()
+
+    return (
+      <div className="edit-temp-var">
+        <div className="edit-temp-var--header">
+          <h1 className="page-header__title">Edit Template Variable</h1>
+          <div className="edit-temp-var--header-controls">
+            <button
+              className="btn btn-default"
+              type="button"
+              onClick={onCancel}
+            >
+              Cancel
+            </button>
+            <button
+              className="btn btn-success"
+              type="button"
+              onClick={this.handleSave}
+              disabled={!this.canSave}
+            >
+              {this.isSaving ? 'Saving...' : 'Save'}
+            </button>
+          </div>
+        </div>
+        <div className="edit-temp-var--body">
+          <div className="edit-temp-var--body-row">
+            <div className="form-group name">
+              <label>Name</label>
+              <input
+                type="text"
+                className="form-control"
+                value={nextTemplate.tempVar}
+                onChange={this.handleChangeName}
+                onKeyUp={this.handleNameKeyUp}
+                onBlur={this.formatName}
+              />
+            </div>
+            <div className="form-group template-type">
+              <label>Type</label>
+              <Dropdown
+                items={TEMPLATE_TYPES_LIST}
+                onChoose={this.handleChooseType}
+                selected={this.dropDownSelection}
+                buttonSize=""
+              />
+            </div>
+          </div>
+          <div className="edit-temp-var--body-row">
+            <TemplateBuilder
+              template={nextTemplate}
+              source={source}
+              onUpdateTemplate={this.handleUpdateTemplate}
+            />
+          </div>
+          <ConfirmButton
+            text={this.isDeleting ? 'Deleting...' : 'Delete'}
+            confirmAction={this.handleDelete}
+            type="btn-danger"
+            size="btn-xs"
+            customClass="delete"
+            disabled={isNew || this.isDeleting}
+          />
+        </div>
+      </div>
+    )
+  }
+
+  private getTemplateBuilder(): ComponentClass<TemplateBuilderProps> {
+    const {
+      nextTemplate: {type},
+    } = this.state
+
+    const component = TEMPLATE_BUILDERS[type]
+
+    if (!component) {
+      throw new Error(`Could not find template builder for type "${type}"`)
+    }
+
+    return component
+  }
+
+  private handleUpdateTemplate = (nextTemplate: Template): void => {
+    this.setState({nextTemplate})
+  }
+
+  private handleChooseType = ({type}) => {
+    const {
+      nextTemplate: {id},
+    } = this.state
+
+    const nextNextTemplate = {...DEFAULT_TEMPLATES[type](), id}
```

Comment:
lol :) you could also just call this `nextTemplate` and use `this.setState({nextTemplate})`. it is funny tho, so i'm down for whatever